store/store.go

```diff
@@ -477,13 +477,18 @@ func (bs *BlockStore) SaveBlock(block *types.Block, blockParts *types.PartSet, s
 	}
 
 	// Save seen commit (seen +2/3 precommits for block)
-	// NOTE: we can delete this at a later height
 	pbsc := seenCommit.ToProto()
 	seenCommitBytes := mustEncode(pbsc)
 	if err := batch.Set(seenCommitKey(height), seenCommitBytes); err != nil {
 		panic(err)
 	}
 
+	// remove the previous seen commit that we have just replaced with the
+	// canonical commit
+	if err := batch.Delete(seenCommitKey(height - 1)); err != nil {
```

Comment:
what if it's the first block? will this panic?

Reply:
My understanding is that it's not detrimental if you try to delete a key that doesn't exist but I will go and make sure that we have a test that covers this case

Reply:
Regadless, we should consider logging instead of a panic perhaps?

Reply:
So, first of all the blockstore has no logger, secondly it doesn't return errors at all. It silences any problems that the database may have or just panics.

Generally the tendermint databases are a bit all over the place.

- The evidence pool has a logger and returns errors for the evidence reactor to handle
- The state store has no logger but will return errors
- The blocks store has no logger and doesn't return errors

It's on my periphery to address this incongruence

Reply:
Agreed, I think it just seems unnecessary to panic in this particular case, since it's not essential to the correctness or liveness of the protocol.